src/net.cpp

```diff
@@ -1544,6 +1548,41 @@ bool CConnman::GenerateSelectSet(std::set<SOCKET> &recv_set, std::set<SOCKET> &s
     return !recv_set.empty() || !send_set.empty() || !error_set.empty();
 }
 
+#ifdef USE_KQUEUE
+void CConnman::SocketEventsKqueue(std::set<SOCKET> &recv_set, std::set<SOCKET> &send_set, std::set<SOCKET> &error_set, bool fOnlyPoll)
+{
+    const size_t maxEvents = 64;
+    struct kevent events[maxEvents];
+
+    struct timespec timeout;
+    timeout.tv_sec = fOnlyPoll ? 0 : SELECT_TIMEOUT_MILLISECONDS / 1000;
+    timeout.tv_nsec = (fOnlyPoll ? 0 : SELECT_TIMEOUT_MILLISECONDS % 1000) * 1000 * 1000;
+
+    wakeupSelectNeeded = true;
+    int n = kevent(kqueuefd, nullptr, 0, events, maxEvents, &timeout);
+    wakeupSelectNeeded = false;
+    if (n == -1) {
+        LogPrintf("kevent wait error\n");
+    } else if (n > 0) {
```

Comment:
Can you document these magic numbers for n?

Reply:
The kevent() system call returns the number of events placed in the
eventlist, up to the value given by nevents.  If an error occurs while
processing an element of the changelist and there is enough room in the
eventlist, then the event will be placed in the eventlist with EV_ERROR
set in flags and the system error in data.  Otherwise, -1 will be
returned, and errno will be set to indicate the error condition.  If the
time limit expires, then kevent() returns 0.

Reply:
All of this is documented in [kevent docs](https://www.freebsd.org/cgi/man.cgi?kevent)